client/landing/stepper/declarative-flow/internals/steps-repository/use-my-domain/index.tsx

```diff
@@ -98,6 +98,7 @@ const UseMyDomain: Step = function UseMyDomain( { navigation, flow } ) {
 					onNextStep={ handleOnNext }
 					isStepper
 					stepLocation={ location }
+					errorAction={ handleGoBack }
```

Comment:
Overall, this is working, but the flow is a bit weird -- I've just searched for something, but the search is not being sent back to the domain page. We should also try to update the domain query value based on what the user was searching for. I think we may be able to do that by updating the `domainForm` value in the `ONBOARD` store.

Reply:
> Overall, this is working, but the flow is a bit weird -- I've just searched for something, but the search is not being sent back to the domain page. We should also try to update the domain query value based on what the user was searching for. I think we may be able to do that by updating the domainForm value in the ONBOARD store.

I wanted to do that initially, but I had no success. I tried updating the `domainForm` value, but that didn't seem to work. Can I ask you for some pointers?

Reply:
I looked into this more deeply, and the issue stems from the `unified-domains` step using locally-persisted state to store and manage the data. I think that unwinding that will need to be tackled in a follow-up PR, as the structure of the `unified-domains` step doesn't really allow for any values to be passed in AFAICT.

https://github.com/Automattic/wp-calypso/blob/ef7a9584bc50a94485bd05376da4382073388df5/client/landing/stepper/declarative-flow/internals/steps-repository/unified-domains/index.tsx#L90-L91